test/integration/apiserver/field_validation_test.go

```diff
@@ -2893,3 +2978,541 @@ func setupCRD(t *testing.T, config *rest.Config, apiGroup string, schemaless boo
+			opts: metav1.CreateOptions{
+				FieldValidation: "Warn",
+			},
+			bodyBase:    validBodyYAML,
+			contentType: "application/yaml",
+		},
+		{
+			name:        "post-no-validation-yaml",
+			bodyBase:    validBodyYAML,
+			contentType: "application/yaml",
+		},
+		{
+			name: "post-ignore-validation-yaml",
+			opts: metav1.CreateOptions{
+				FieldValidation: "Ignore",
+			},
+			bodyBase:    validBodyYAML,
+			contentType: "application/yaml",
+		},
+	}
+
+	for _, bm := range benchmarks {
+		b.Run(bm.name, func(b *testing.B) {
+			b.ResetTimer()
+			b.ReportAllocs()
+			for n := 0; n < b.N; n++ {
+				body := []byte(fmt.Sprintf(bm.bodyBase, fmt.Sprintf("test-deployment-%s-%d-%d-%d", bm.name, n, b.N, time.Now().UnixNano())))
+				req := client.CoreV1().RESTClient().Post().
+					AbsPath("/apis/apps/v1").
+					Namespace("default").
+					Resource("deployments").
+					SetHeader("Content-Type", bm.contentType).
+					VersionedParams(&bm.opts, metav1.ParameterCodec)
+				result := req.Body(body).Do(context.TODO())
+				if result.Error() != nil {
+					b.Fatalf("unexpected request err: %v", result.Error())
+				}
+			}
+		})
+	}
+}
+
+func benchFieldValidationPut(b *testing.B, client clientset.Interface) {
+	var testcases = []struct {
+		name        string
+		opts        metav1.UpdateOptions
+		putBodyBase string
+		contentType string
+	}{
+		{
+			name: "put-strict-validation",
+			opts: metav1.UpdateOptions{
+				FieldValidation: "Strict",
+			},
+			putBodyBase: invalidBodyJSON,
+		},
+		{
+			name: "put-warn-validation",
+			opts: metav1.UpdateOptions{
+				FieldValidation: "Warn",
+			},
+			putBodyBase: invalidBodyJSON,
+		},
+		{
+			name: "put-default-ignore-validation",
+			opts: metav1.UpdateOptions{
+				FieldValidation: "Ignore",
+			},
+			putBodyBase: invalidBodyJSON,
+		},
+		{
+			name:        "put-ignore-validation",
+			putBodyBase: invalidBodyJSON,
+		},
+		{
+			name: "put-strict-validation-yaml",
+			opts: metav1.UpdateOptions{
+				FieldValidation: "Strict",
+			},
+			putBodyBase: invalidBodyYAML,
+			contentType: "application/yaml",
+		},
+		{
+			name: "put-warn-validation-yaml",
+			opts: metav1.UpdateOptions{
+				FieldValidation: "Warn",
+			},
+			putBodyBase: invalidBodyYAML,
+			contentType: "application/yaml",
+		},
+		{
+			name: "put-ignore-validation-yaml",
+			opts: metav1.UpdateOptions{
+				FieldValidation: "Ignore",
+			},
+			putBodyBase: invalidBodyYAML,
+			contentType: "application/yaml",
+		},
+		{
+			name:        "put-no-validation-yaml",
+			putBodyBase: invalidBodyYAML,
+			contentType: "application/yaml",
+		},
+	}
+
+	for _, tc := range testcases {
+		b.Run(tc.name, func(b *testing.B) {
+			b.ResetTimer()
+			b.ReportAllocs()
+			for n := 0; n < b.N; n++ {
+				deployName := fmt.Sprintf("%s-%d-%d-%d", tc.name, n, b.N, time.Now().UnixNano())
+				postBody := []byte(fmt.Sprintf(string(validBodyJSON), deployName))
+
+				if _, err := client.CoreV1().RESTClient().Post().
```

Comment:
I'm guessing that's how we're doing it in the other existing tests, but shouldn't we first loop and run all the posts, then reset the timer and run the Put if that's what we actually want to test/bench? (And I'm guessing that comment applies to other methods below)

Reply:
Yea, good catch, fixed